src/olympia/addons/views.py

```diff
@@ -812,33 +813,127 @@ def as_view(cls, **initkwargs):
-                    self.request).get_value()
+                value = AddonAppVersionQueryParam(self.request).get_values()
+                appversions = {
+                    'min': value[1],
+                    'max': value[2]
+                }
             except ValueError:
-                raise exceptions.ParseError('Invalid app parameter.')
-        return self.application_id
+                raise exceptions.ParseError('Invalid appversion parameter.')
+        else:
+            appversions = None
+
+        # type is optional, unless appversion is set. That's because the way
+        # dicts and language packs have their compatibility info set in the
+        # database differs, so to make things simpler for us we force clients
+        # to filter by type if they want appversion filtering.
+        if AddonTypeQueryParam.query_param in self.request.GET or appversions:
+            try:
+                addon_types = (AddonTypeQueryParam(self.request).get_value(),)
+            except ValueError:
+                raise exceptions.ParseError(
+                    'Invalid or missing type parameter while appversion '
+                    'parameter is set.')
+        else:
+            addon_types = (amo.ADDON_LPAPP, amo.ADDON_DICT)
+        return application, addon_types, appversions
 
     def get_queryset(self):
-        types = (amo.ADDON_DICT, amo.ADDON_LPAPP)
-        return Addon.objects.public().filter(
-            appsupport__app=self.get_application_id(), type__in=types,
-            target_locale__isnull=False).exclude(target_locale='')
+        application, addon_types, appversions = self.get_query_params()
+        if addon_types == (amo.ADDON_LPAPP,) and appversions:
+            return self.get_language_packs_queryset_with_appversions(
+                application, appversions)
+        else:
+            # appversions filtering only makes sense for language packs only,
+            # so it's ignored here.
+            return self.get_queryset_base(application, addon_types)
+
+    def get_queryset_base(self, application, addon_types):
+        return (
+            Addon.objects.public()
+                 .no_cache()
+                 .filter(appsupport__app=application, type__in=addon_types,
+                         target_locale__isnull=False)
+                 .exclude(target_locale='')
+            # Deactivate default transforms which fetch a ton of stuff we
+            # don't need here like authors, previews or current version.
+            # It would be nice to avoid translations entirely, because the
+            # translations transformer is going to fetch a lot of translations
+            # we don't need, but some language packs or dictionaries have
+            # custom names, so we can't use a generic one for them...
+                 .only_translations()
+            # Since we're fetching everything with no pagination, might as well
+            # not order it.
+                 .order_by()
```

Comment:
No order?  Sad!  I kind of like the results being ordered by locale code or something sensible.  Unless it's a huge perf hit, of course.

Reply:
Thing is, there were ordered by `created` date before because nothing was set. I figured it was almost useless, and not worth the extra work for the database (it's indexed, but still)

Reply:
I don't care really.